File: backend/internal/domain/master.go
package domain

import "context"

type Master struct {
	ID         int64   `db:"id" json:"id"`
	Name       string  `db:"name" json:"name"`
	PhotoURL   string  `db:"photo_url" json:"photo"`
	Rating     float64 `db:"rating" json:"rating"`
	Level      string  `db:"level" json:"level"`
	ProgramIDs []int64 `json:"programIds"`
}

type MasterRepository interface {
	QueryMasters(ctx context.Context) ([]Master, error)
	GetMasterByID(ctx context.Context, id int64) (Master, error)
}

File: mobile/lib/core/models/master.dart
class Master {
  final int id;
  final String name;
  final String photo;
  final double rating;
  final String level;
  final List<int> programIds;

  const Master({
    required this.id,
    required this.name,
    required this.photo,
    required this.rating,
    required this.level,
    required this.programIds,
  });

  factory Master.fromJson(Map<String, dynamic> json) => Master(
    id: json['id'] as int,
    name: json['name'] as String,
    photo: json['photo'] as String,
    rating: (json['rating'] as num).toDouble(),
    level: json['level'] as String,
    programIds: (json['programIds'] as List).cast<int>(),
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'photo': photo,
    'rating': rating,
    'level': level,
    'programIds': programIds,
  };
}
